src/internal/cmdutil/env.go

```diff
@@ -71,10 +71,10 @@ func populateInternal(reflectValue reflect.Value, decoderMap map[string]string,
 	if reflectValue.Type().Kind() != reflect.Struct {
 		return errors.Errorf("%s: %v", expectedStructErr, reflectValue.Type())
 	}
-	numField := reflectValue.NumField()
-	for i := 0; i < numField; i++ {
+	for i := 0; i < reflectValue.NumField(); i++ {
 		structField := reflectValue.Type().Field(i)
-		if structField.Type.Kind() == reflect.Struct {
+		ptrToStruct := structField.Type.Kind() == reflect.Ptr && structField.Type.Elem().Kind() == reflect.Struct
+		if structField.Type.Kind() == reflect.Struct || ptrToStruct {
```

Comment:
I'm a little confused by the code before - from what I can tell, it would never run on the `serviceenv.Configuration` object because all of its subfields are pointers to structs, but I'm pretty sure this would have broken `pachd` if it didn't work...